### PR TITLE
Refactor: analytics for `claims_provider`

### PR DESCRIPTION
**Pull request by thekaveman**
Part of #2247 

Included a number of small renames/docs updates that I missed during #2379.

### CODE REVIEW

File: benefits/oauth/analytics.py [hunk 1]
```diff
@@ -10,9 +10,9 @@ class OAuthEvent(core.Event):
 
     def __init__(self, request, event_type):
         super().__init__(request, event_type)
-        verifier = session.flow(request)
-        if verifier and verifier.uses_claims_verification:
-            self.update_event_properties(auth_provider=verifier.claims_provider.client_name)
+        flow = session.flow(request)
+        if flow and flow.uses_claims_verification:
+            self.update_event_properties(claims_provider=flow.claims_provider.client_name)
```
Comment (thekaveman) [inline benefits/oauth/analytics.py:15]: This change was the actual point of this PR

**Review by angela-tran — APPROVED** [review]
This looks good to me! Thanks for catching those old variable names / docs that needed to be updated from `verifier` to `flow`. I did a global search of "auth_provider", "auth provider", and "verifier" to double-check it as well.

Question (maybe more for #2247 than this PR): will there be commits in cal-itp/data-infra for updating the old references to `auth_provider`? e.g. in the [SQL](https://github.com/cal-itp/data-infra/pull/3468/files#diff-5ab4887f3fabbef14579bf85ff73b47494e44dcd7f23b8355680d60fb8bf1fbcR37), YAML, etc.

**Comment by thekaveman** [comment]
@angela-tran

> will there be commits in cal-itp/data-infra for updating the old references

Yes, see https://github.com/cal-itp/data-infra/pull/3468 -- I figured we could just use one big PR over there to make all these changes once we're done on the Benefits side. Definitely open to smaller PRs for each individual fix too, just didn't want to burden that team with a ton of PR/reviews.

I'm going to wait to merge this until @lalver1 merges #2402 since there will be conflicts, and it will probably be easier to resolve in this PR.

**Review by angela-tran — APPROVED** [review]
👍

**Pull request merged**